execution-engine/src/hcall/wasmi.rs

```diff
@@ -529,30 +513,30 @@ impl WasmiHostProvisioningState {
         let address: u32 = args.nth(1);
 
         match self.get_memory() {
-            None => Err(FatalHostError::NoMemoryRegistered),
-            Some(memory) => match self.get_current_data_source(index as usize) {
-                None => return Ok(VeracruzError::BadInput),
-                Some(frame) => {
-                    let result = frame.get_data().len() as u32;
-                    let result: Vec<u8> = result.to_le_bytes().to_vec();
-
-                    if let Err(_) = memory.set(address, &result) {
-                        return Err(FatalHostError::MemoryWriteFailed {
-                            memory_address: address as usize,
-                            bytes_to_be_written: result.len() as usize,
-                        });
-                    }
+            None => Err(FatalEngineError::NoMemoryRegistered),
+            Some(memory) => 
+            {
+                //TODO FILL IN principal
+                let result = self.read_file(&self.program,&format!("input-{}",index))?.ok_or(format!("File input-{} cannot be found",index))?.len() as u32;
```

Comment:
Has `rustfmt` been run on these files?  There's some bad spacing/layout here.

Reply:
I will run a rustfmt once everything is good. As it is easy for git to check the diff without fmt change,